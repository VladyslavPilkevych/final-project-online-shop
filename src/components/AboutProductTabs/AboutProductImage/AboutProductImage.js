/* eslint-disable max-len */
/* eslint-disable jsx-a11y/interactive-supports-focus */
import React from 'react';
import PropTypes from 'prop-types';
import Slider from 'react-slick';
import { useDispatch } from 'react-redux';
import { addToCart } from '../../../store/actionCreators/cartAC';
import addToFavIcon from '../../../assets/images/addToFavIcon.png';
import styles from './AboutProductImage.module.scss';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';

function AboutProductImage(props) {
  const { urls, model, id } = props;
  const dispatch = useDispatch();
  const settings = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1,
  };
  console.log(urls);
  return (
    <div className={styles.detailsWrapper}>
      <div role="button" onClick={() => { console.log('Add to Wishlist', id); }} className={styles.addToCartIconWrapper}>
        <img
          src={addToFavIcon}
          alt="Add to Cart Icon"
          className={styles.addToCartIcon}
        />
      </div>
      <div className={styles.imageWrapper}>
        <Slider {...settings}>
          {/* <div><img src={urls[0]} alt={model} className={styles.imageMain} /></div> */}
          {/* <div><img src={urls[1]} alt={model} className={styles.imageMain} /></div> */}
          {urls && urls.map((elem) => <div key={model}><img src={`${elem}`} alt={model} className={styles.imageMain} /></div>)}

        </Slider>
      </div>
    </div>
  );
}

AboutProductImage.propTypes = {
  id: PropTypes.string,
  urls: PropTypes.array,
  model: PropTypes.string,
};

AboutProductImage.defaultProps = {
  id: '',
  urls: [],
  model: '',
};

export default AboutProductImage;
