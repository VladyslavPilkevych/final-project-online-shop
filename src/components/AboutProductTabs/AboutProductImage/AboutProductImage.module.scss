.imageSideWrapper {
  width: 80%;
}

.imageMain {
  width: 100%;
  max-height: 350px !important;
}

.detailsWrapper {
  display: flex;
}

.addToCartIconWrapper {
  cursor: pointer;
  margin: 30px 53px 0 23px;
}

.imgmin {
  width: 30px;
  // height: 30px;
}

ul {
  display: flex !important;
  justify-content: space-between;
  align-items: center;
}

ul.slick-dots li {
  height: 50px;
}

.slick-track {
  display: flex !important;
  max-height: 350px !important;
}
.slick-list {
  max-height: 350px !important;
}
