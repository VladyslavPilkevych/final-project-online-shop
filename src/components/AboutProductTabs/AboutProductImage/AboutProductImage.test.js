import { render } from '@testing-library/react';
import AboutProductImage from './AboutProductImage';

describe('Snapshot AboutProductImage should render', () => {
  test('Snapshot AboutProductImage is rendered', () => {
    const { asFragment } = render(<AboutProductImage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
