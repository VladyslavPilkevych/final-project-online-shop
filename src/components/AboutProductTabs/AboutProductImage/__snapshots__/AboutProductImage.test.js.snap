// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot AboutProductImage should render Snapshot AboutProductImage is rendered 1`] = `
<DocumentFragment>
  <div
    class="imageSideWrapper"
  >
    <div
      class="imageWrapper"
    >
      <div
        class="slick-slider slick-initialized"
      >
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
